src/serialize.rs

```diff
@@ -0,0 +1,103 @@
+use super::*;
+use serde::{Serialize, de::DeserializeOwned};
+
+#[cfg(test)]
+mod test{
+    const ACTION_SELECT: &str = "SELECT";
+    const ACTION_SUPER_SELECT: &str = "SUPER_SELECT";
+    const ACTION_AWESOME_SUPER_SELECT: &str = "AWESOME_SUPER_SELECT";
+    const TESTPATH : &str = "test.keymap";
+    use bevy::prelude::*;
+    use super::*;
+    #[derive(Hash, PartialEq, Eq, Clone, Serialize, serde::Deserialize, Debug)]
+    enum TestAction {
+        Select,
+        SuperSelect,
+        AwesomeSuperSelect,
+    }
+    fn test_binding<T>(to_test : T) where T : Into<Binding>{
+        let binding : Binding = to_test.into();
+        let serialized = ron::to_string(&binding).expect("Failed to serialize binding");
+        let deserialized : Binding = ron::from_str(&serialized).expect("Failed to deserialize binding");
+        assert_eq!(binding, deserialized);
+    }
+    
+    
+    fn test_action(action : Action){
+        let serialized = ron::to_string(&action).expect("Failed to serialize action");
+        let deserialized : Action = ron::from_str(&serialized).expect("Failed to deserialize action");
+        assert_eq!(action, deserialized)
+    }
+    #[test]
+    fn test_inputmap_string(){
+        let mut map = super::InputMap::<String>::default();
+        map.bind(ACTION_SELECT, KeyCode::Return)
+        .bind(ACTION_SELECT, GamepadButtonType::South)
+        .bind(ACTION_SUPER_SELECT, vec![KeyCode::LAlt, KeyCode::Return])
+        .bind(ACTION_SUPER_SELECT, vec![KeyCode::RAlt, KeyCode::Return])
+        // This should bind left/right control and left/right alt, but the combos would get ridiculous so hopefully this is sufficient.
+        .bind(
+            ACTION_AWESOME_SUPER_SELECT,
+            vec![KeyCode::LAlt, KeyCode::LControl, KeyCode::Return],
+        );
+        map.save_to_path(TESTPATH).expect("Failed to save string InputMap");
```

Comment:
```suggestion
        map.save_to_path(TEST_PATH).expect("Failed to save string InputMap");
```

Also, should the other test save to `TEST_PATH` as well?